rasa/nlu/featurizers/dense_featurizer/mitie_featurizer.py

```diff
@@ -133,3 +131,26 @@ def features_for_tokens(
             vec = vec / len(tokens)
 
         return np.expand_dims(vec, axis=0)
+
+    def features_for_tokens(
+        self,
+        tokens: List[Token],
+        feature_extractor: "mitie.total_word_feature_extractor",
+    ) -> np.ndarray:
+        cls_token_used = tokens[-1].text == CLS_TOKEN if tokens else False
+
+        input_tokens = tokens
+        if cls_token_used:
+            input_tokens = tokens[:-1]
+
+        if self.return_sequence:
+            features = self._features_as_sequence(input_tokens, feature_extractor)
+        else:
+            features = self._features_as_non_sequence(input_tokens, feature_extractor)
+
```

Comment:
it looks like, we do double job here, I think it makes sense to always calculate features, then either return or concat the mean